api/utils.spec.ts

```diff
@@ -0,0 +1,42 @@
+import { expect } from 'chai';
+import { allDoNotContainDangerousCharacters, hasNoDangerousCharacters, urlHasNoDangerousCharacters } from './utils';
+
+describe('api utils', () => {
+  describe('hasNoDangerousCharacters', () => {
```

Comment:
please add tests for dangerous chars

Reply:
There are tests for dangerous characters already - do you mean more tests testing more of the dangerous characters?